docs/zero_knowledge.md

```diff
@@ -1,136 +1,76 @@
-This module explains the concepts behind Zero-Knowledge proofs
+# Zero-Knowledge Proofs
 
-# Zero-Knowledge Proof
-
-In the following article you will learn about the definition of a 
-Zero-Knowledge proof system, understand what that definition implies and 
-ultimately learn about different Zero-Knowledge protocols.
+In the following article, you will learn about Zero-Knowledge Proofs, understand what they imply, and ultimately learn about different Zero-Knowledge Proof constructions.
 
 ## Definition
 
-A Zero-Knowledge system is a proving-protocol that satisfies the following 
-three properties:
+A Zero-Knowledge Proof (ZKP) [[1]](#1) is a cryptographic primitive which allows a proving party $P$ to convince a verifying party $V$ that a statement is true, without leaking any secret information. A statement $u$ is a set of elements known by both parties, and the secret information only known by $P$ is the witness $w$. Formally speaking, ZKPs must satisfy 3 properties:
 
-1. **Completeness**  
-  Every prover with a valid solution will be able to convince a verifier
-2. **Soundness**  
-  Only a prover with a valid solution can convince a verifier
-3. **Zero-Knowledge**  
-  The prover proofs they know a secret without revealing any information 
-  about the secret itself
+- **Completeness:** If the statement is true, $P$ must be able to convince $V$.
+- **Soundness** If the statement is false, $P$ must not be able to convince $V$ that the statement is true, except with negligible probability.
+- **Zero-Knowledge** $V$ must not learn any information from the proof beyond the fact that the statement is true.
 
 ## Example
+To understand what the above definition means, let us look at an example.
+
+Consider the graph [coloring problem][graph coloring]. You have a graph of vertices connected by edges. Every edge connects to two vertices and every vertex can be connected to any number of edges.  Now you get three colors and need to color each vertex in such a way, that no edge connects to two vertices of the same color. Not all graphs have a valid solution to this problem and it happens that while it is easy to construct a graph with a valid solution, it is not easy at all to determine *if* a given graph has a valid solution or *how* that solution might look like.
+
+<p align=center>
+<img src="https://upload.wikimedia.org/wikipedia/commons/c/c2/Triangulation_3-coloring.svg" alt="drawing" width="35%"/>
+</p>
+
```

Comment:
What is the reason you didn't use the markdown way of including the picture of the graph? Did it not render?

Reply:
I wanted to center it and also to specify the size, to fit in different screens, and using html tags was the solution I found for it... if you know an alternative, let me know :)